server/src/session.ts

```diff
@@ -166,19 +166,17 @@ export class Session {
   }
 
   private onDidOpenTextDocument(params: lsp.DidOpenTextDocumentParams) {
-    const {uri, languageId} = params.textDocument;
+    const {uri, languageId, text} = params.textDocument;
     const filePath = uriToFilePath(uri);
     if (!filePath) {
       return;
     }
     const scriptKind = languageId === LanguageId.TS ? ts.ScriptKind.TS : ts.ScriptKind.External;
-    // Since we are only keeping track of changes of files that are already on
-    // disk (see documentSelector.scheme in extension.ts), the content on disk
-    // is up-to-date when a file is first opened in the editor.
-    // In this case, we should not pass fileContent to projectService.
-    const fileContent = undefined;
     try {
-      const result = this.projectService.openClientFile(filePath, fileContent, scriptKind);
+      // The content could be newer than that on disk. This could be due to
+      // buffer in the user's editor which has not been saved to disk.
+      // See https://github.com/angular/vscode-ng-language-service/issues/632
+      const result = this.projectService.openClientFile(filePath, text, scriptKind);
```

Comment:
Will this only ever be called on files that already exist on disk? The other case is that we have a buffer that has never been recorded on disk:

https://github.com/angular/vscode-ng-language-service/blob/3c7f64b4c3805087b8055644f1a7a01fd68a2211/client/src/extension.ts#L28-L33

and in this case, what will happen if we try to open the client file?

Reply:
If a buffer is not on disk, it's an "untitled" document. Since untitled document is not part of the document selector that we specified, we will never receive the `DidOpenTextDocument` event for such buffer.

Reply:
Okay. I think this is a mixup that caused the change in 476. LGTM